src/manual/containers-async.adoc

```diff
@@ -82,4 +87,57 @@ The async container also supports success and error callback actions. This is an
 </testcase>
 ----
 
-So you can add test actions that get executed based on the async test action outcome `success` or `error`.
\ No newline at end of file
+So you can add test actions that get executed based on the async test action outcome `success` or `error`.
+
+If you are using this container to send or receive messages, you have to use the unique correlation id of the
+message to link the actions concerning this message. Otherwise the testcase might associate a send or receive action
+with the wrong message. Please note that it is **not** required to pass this id through your system under test. The
+assignment of the `MessageHeaders.ID` is done internally. Only the mapping between the request and response has to be
+done by the author of the test. As you can see in the following example, the value of the header `MessageHeaders.ID` is
+stored in the variable `request#1` respectively `request#2`. This variable is reused in the receive action to identify
+the correct response from the server.
+
+.Java DSL
+[source,java]
+----
+@CitrusTest
+public void testAsync() {
+
+    async().actions(
+        http(b -> b.client(httpClient)
+                .send()
+                .post("/foo")
+                .extractFromHeader(MessageHeaders.ID, "request#1")
+                .payload("{ \"info\": \"foo\"}")),
+
+        //SUT echoing the input
+
+        http(b -> b.client(httpClient)
+                .receive()
+                .response(HttpStatus.OK)
+                .payload("{ \"info\": \"foo\"}")
+                .selector(
+                    Collections.singletonMap(
+                        MessageHeaders.ID,
+                        "${request#1}")))
+
+    );
+
+    async().actions(
+        http(b -> b.client(httpClient)
+                .send()
+                .post("/boo")
+                .extractFromHeader(MessageHeaders.ID, "request#2")
+                .payload("{ \"info\": \"boo\"}")),
+
```

Comment:
Hello? Echo? 😃
```suggestion

        //SUT echoing the input
```

Reply:
Just to clarify the semantics of the test here. 😅
Don't you like it? 🤣

Reply:
I sure **do** like explicit / clear examples! It was just missing on the second example. 👍

Reply:
Ah, sorry haven't seen this addition on my mobile device :wink: